### PR TITLE
JDK-8325037: x86: enable and fix hotspot/jtreg/compiler/vectorization/TestRoundVectFloat.java

### PR DESCRIPTION
**Pull request by Hamlin-Li**
Hi,
Can you help to review this simple patch to fix test TestRoundVectFloat.java?
Thanks!

FYI: This test is not actually tested, need to fix the test applying filter and IR matching rule.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8325037](https://bugs.openjdk.org/browse/JDK-8325037): x86: enable and fix hotspot/jtreg/compiler/vectorization/TestRoundVectFloat.java (**Bug** - P5)

### Reviewers
 * [Tobias Hartmann](https://openjdk.org/census#thartmann) (@TobiHartmann - **Reviewer**)
 * [Vladimir Kozlov](https://openjdk.org/census#kvn) (@vnkozlov - **Reviewer**)
 * [Jatin Bhateja](https://openjdk.org/census#jbhateja) (@jatin-bhateja - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/17649/head:pull/17649` \
`$ git checkout pull/17649`

Update a local copy of the PR: \
`$ git checkout pull/17649` \
`$ git pull https://git.openjdk.org/jdk.git pull/17649/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 17649`

View PR using the GUI difftool: \
`$ git pr show -t 17649`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/17649.diff">https://git.openjdk.org/jdk/pull/17649.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/17649#issuecomment-1918796885)

### CODE REVIEW

**Review by TobiHartmann — CHANGES_REQUESTED** [review]

**Review by TobiHartmann — APPROVED** [review]

**Review by vnkozlov — APPROVED** [review]
Good.

File: test/hotspot/jtreg/compiler/vectorization/TestRoundVectFloat.java [hunk 1]
```diff
@@ -44,13 +44,12 @@ public class TestRoundVectFloat {
...
   public static void main(String args[]) {
       TestFramework.runWithFlags("-XX:-TieredCompilation",
-                                 "-XX:UseAVX=1",
                                  "-XX:CompileThresholdScaling=0.3");
       System.out.println("PASSED");
   }
 
   @Test
-  @IR(applyIf = {"UseAVX", " > 1"}, counts = {"RoundVF" , " > 0 "})
+  @IR(applyIf = {"UseAVX", " > 1"}, counts = {IRNode.ROUND_VF , " > 0 "})
```
[the remark this answers is not in the document]
Comment (jatin-bhateja) [reply test/hotspot/jtreg/compiler/vectorization/TestRoundVectFloat.java:52]: FTR, backend implementation of rounding first adds a broadcasted 0.5 value to incoming floating point vector lanes, register only flavor of broadcast operation is only supported by AVX2 targets. We can broadcast 0.5 into constant table emitted prior to method body to lift this limitation. Another concern is around different max vector size for floating point and integral vectors on AVX1 targets, 256 bit float vector vs 128 bit integral vector. Thus, a sharper check in match_rule_supported_vector (size_in_bits  > 128 && UseAVX < 2) should be added to support this for AVX1 targets as we are emitting an integral compare in special case handling for NaN and -/+0.0 / Inf.

**Review by jatin-bhateja — APPROVED** [review]

**Comment by Hamlin-Li** [comment]
> I'll run this through our testing and report back once it passed.

Thanks for testing, I'll wait for your feedback.

**Comment by Hamlin-Li** [comment]
@TobiHartmann @vnkozlov @jatin-bhateja Thanks for your revewiing!

/integrate

**Pull request closed without merging**